addons/slack/install.js

```diff
@@ -13,64 +13,97 @@ const getTemplateFiles = (fileNames) =>
+              },
+            ],
+            functionPermissions: {
+              allow: [
+                ...(eventsResourcePath
+                  ? [
+                      {
+                        // :execute connector instance
+                        action: 'function:execute',
+                        resource: eventsResourcePath,
+                      },
+                    ]
+                  : []),
+                {
+                  // :post to its own storage
+                  action: 'storage:*',
+                  resource:
+                    '/account/{{accountId}}/subscription/{{subscriptionId}}/storage/boundary/{{boundaryId}}/function/{{functionId}}/',
+                },
+              ],
+            },
+          },
+          metadata: {
+            fusebit: {
+              editor: {
+                navigationPanel: {
+                  hideFiles: ['index.js'],
+                },
+              },
             },
+            ...ctx.body.metadata,
           },
         },
-        ...ctx.body.metadata,
-      },
-    });
-  } catch (e) {
-    // Could not create function, clean up storage
-    await Sdk.deleteStorage(ctx, storageSpecification);
-    throw e;
-  }
-
-  let registrationUrl = ctx.body.configuration.fusebit_events_registration;
-  let token = ctx.body.configuration.fusebit_events_token;
-  if (registrationUrl && token) {
-    try {
-      Sdk.debug('Events API configuration was present');
-      await Superagent.post(registrationUrl)
-        .set('Authorization', `Bearer ${token}`)
-        .send({
-          app: ctx.body.configuration.slack_app_id,
-          team_id: ctx.body.configuration.slack_team_id,
-          handler: (await Sdk.getFunctionUrl(ctx)) + '/events/ingest',
-        });
-
-      Sdk.debug('Events API registration succeeded');
+        ctx.fusebit.functionAccessToken
+      );
     } catch (e) {
-      // Couldn't do Events API registration, clean up function and storage
-      await Sdk.deleteStorage(ctx, storageSpecification);
-      await Sdk.deleteFunction(ctx);
       throw e;
     }
-  }
 
-  return { status: 200, body: { status: 200 } };
+    let registrationUrl = ctx.body.configuration.fusebit_events_registration;
+    if (registrationUrl) {
+      try {
+        Sdk.debug('Events API configuration was present');
+        await Superagent.post(registrationUrl)
+          .set('Authorization', `Bearer ${ctx.fusebit.functionAccessToken}`)
+          .send({
+            app: ctx.body.configuration.slack_app_id,
+            team_id: ctx.body.configuration.slack_team_id,
+            handler: (await Sdk.getFunctionUrl(ctx, ctx.fusebit.functionAccessToken)) + '/events/ingest',
+          });
+
+        Sdk.debug('Events API registration succeeded');
+      } catch (e) {
+        // Couldn't do Events API registration, clean up function and storage
```

Comment:
What's up with the "and storage" part? There is no storage created at this point, is there?

Reply:
Correct, that was a stale comment, fixed